Unity/Champloo/Assets/Scripts/Player/Movement States/Movement Specials/OnSlide.cs

```diff
@@ -0,0 +1,160 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class OnSlide : OnMovementSpecial
+{
+
+    [SerializeField]
+    private float wallSlideUpForce;
+    [SerializeField]
+    private float wallSlideDownForce;
+    [SerializeField]
+    private float wallSlideUpDecay;
+    [SerializeField]
+    private float wallSlideDownDecay;
+    [SerializeField]
+    private Vector2 groundSlideForce;
+    [SerializeField]
+    private float groundSlideDecay;
+
+    [SerializeField]
+    private PlayRandomSource wallSlideSound;
+
+    private bool goingRight;
```

Comment:
I'm not sure why you have this as a class level variable; both instances when you use it end up resetting it each call